src/zabapgit_object_ucsa.prog.abap

```diff
@@ -0,0 +1,240 @@
+*&---------------------------------------------------------------------*
+*& Include zabapgit_object_ucsa
+*&---------------------------------------------------------------------*
+
+CLASS lcl_object_ucsa DEFINITION INHERITING FROM lcl_objects_super FINAL.
+
+  PUBLIC SECTION.
+    INTERFACES lif_object.
+
+  PRIVATE SECTION.
+    TYPES:
+      ty_id TYPE c LENGTH 30.
+
+    METHODS:
+      get_persistence
+        IMPORTING
+          iv_id                 TYPE ty_id
+        RETURNING
+          VALUE(ro_persistence) TYPE REF TO object.
+
+ENDCLASS.
+
+CLASS lcl_object_ucsa IMPLEMENTATION.
+
+  METHOD lif_object~has_changed_since.
+
+    rv_changed = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~changed_by.
+
+    rv_user = c_user_unknown.
+
+  ENDMETHOD.
+
+  METHOD lif_object~get_metadata.
+
+    rs_metadata = get_metadata( ).
+    rs_metadata-delete_tadir = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~exists.
+
+    DATA: lv_id          TYPE ty_id,
+          lo_persistence TYPE REF TO object.
+
+    lv_id = ms_item-obj_name.
+
+    TRY.
+        lo_persistence = get_persistence( lv_id ).
+
+        CALL METHOD lo_persistence->('IF_UCON_SA_PERSIST~LOAD')
+          EXPORTING
+            version  = 'A'
+            language = sy-langu.
+
+      CATCH cx_root.
+        rv_bool = abap_false.
+        RETURN.
+    ENDTRY.
+
+    rv_bool = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~serialize.
+
+    DATA: lv_id          TYPE ty_id,
+          lx_root        TYPE REF TO cx_root,
+          lv_text        TYPE string,
+          lo_persistence TYPE REF TO object,
+          lr_sa          TYPE REF TO data.
+
+    FIELD-SYMBOLS: <sa>     TYPE any,
```

Comment:
I think you can rename all instances of `sa` to a more meaningful name. What does SA stand for?

Reply:
changed to complete_comm_assembly